pyexchange/fix.py

```diff
@@ -112,7 +112,9 @@ async def _read_message(self):
 
             # Handle session messages, queue application messages.
             if not self._handle_session_message(message):
+                await self.lock.acquire()
                 self._handle_application_message(message)
+                self.lock.release()
```

Comment:
Recommend calling `_handle_application_message` in a `try` and releasing the lock in a `finally`.  A catch isn't needed.

Reply:
Thanks for the suggestion, updated accordingly